packages/admin-ui/src/table/components/table-cell.tsx

```diff
@@ -2,23 +2,54 @@ import { createComponent, useElement } from '@vtex/admin-ui-react'
 import type { VariantProps } from '@vtex/admin-ui-core'
 import * as styles from '../styles/table-cell.styles'
 import type { TableColumn } from '../types'
+import React, { useRef, useEffect } from 'react'
+import { useStateContext } from '../context'
+import { Box } from '../../box'
 
 export const TableCell = createComponent<'div', CellOptions>((props) => {
-  const { column, onClick, role = 'cell', ...cellProps } = props
+  const { column, onClick, role = 'cell', ref, children, ...cellProps } = props
+  const { columns } = useStateContext()
 
   const clickable = !!onClick
 
+  const isFixed = !!column?.fixed
+
+  const lastFixedColumn = columns.filter((col) => !!col?.fixed).slice(-1)[0]
+
+  const belongsToLastFixedColumn = isFixed && lastFixedColumn === column
+
+  const cellRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (cellRef.current && isFixed) {
+      cellRef.current.style.left = `${cellRef.current.offsetLeft}px`
+    }
+  }, [])
+
   return useElement('div', {
+    ref: cellRef as any,
+    className: isFixed ? '__admin-ui-fixed-cell' : '',
```

Comment:
I think you can take the `className` from props and use it this way:
```suggestion
import { cx } from '@vtex/admin-ui-core'
...
    className: isFixed ? cx('__admin-ui-fixed-cell', className) : className,
```